src/shared/helpers/EventHelper.ts

```diff
@@ -43,17 +45,26 @@ export default class EventHelper {
     const newNotificationTypes = subscriptionState.optedOut ? -2 : 1;
     await context.subscriptionManager.updatePushSubscriptionNotificationTypes(newNotificationTypes);
 
+    const currentPushToken = await MainHelper.getCurrentPushToken();
+    const pushModel: OSModel<SubscriptionModel> = await OneSignal.coreDirector.getCurrentPushSubscriptionModel();
+    const pushSubscriptionId = pushModel.data?.id;
+
     await Database.setIsPushEnabled(isPushEnabled);
     appState.lastKnownPushEnabled = isPushEnabled;
+    appState.lastKnownPushToken = currentPushToken;
+    appState.lastKnownPushId = pushSubscriptionId;
     await Database.setAppState(appState);
 
-    // TO DO: should use id and token. will require more changes to save previous state
     const change: SubscriptionChangeEvent = {
       previous: {
-        enabled: !isPushEnabled,
+        id: appState.lastKnownPushId,
+        token: appState.lastKnownPushToken,
+        optedIn: lastKnownPushEnabled,
```

Comment:
we're not surfacing distinction of native permission and the preference.

check the native permission & the preference = optedIn setting

`lastKnownOptedIn`